### PR TITLE
Lint codebase

### PR DESCRIPTION
**Pull request by XhmikosR**
* remove unused variables
* remove unneeded else clauses
* use `const` when possible
* use template liters
* use spread
* throw a `TypeError` instead of a generic error
* use `forEach` instead of `map` since we don't return anything
* remove trailing commas and add missing semicolons

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -79,8 +75,8 @@ class ReaddirpStream extends Readable {
   static get defaultOptions() {
     return {
       root: '.',
-      fileFilter: path => true,
-      directoryFilter: path => true,
+      fileFilter: () => true,
```
Comment (XhmikosR) [inline index.js]: I'm not sure about these 2

File: index.js [hunk 2]
```diff
@@ -154,8 +150,7 @@ class ReaddirpStream extends Readable {
 
     if (this.destroyed) return;
 
-    for (let i = 0; i < entries.length; i++) {
-      const entry = entries[i];
+    for (const entry of entries) {
```
Comment (XhmikosR) [inline index.js]: I switched to a for of loop since you already use a for of loop earlier

File: test.js [hunk 3]
```diff
@@ -21,36 +25,36 @@ let testCount = 0;
 let currPath;
 
 const read = async (options) => {
-  return await readdirp.promise(currPath, options);
```
Comment (XhmikosR) [inline test.js]: This is redundant

File: index.js (hunk 1 above)
Comment (paulmillr) [reply index.js]: Leave it as before because typescript

**Pull request merged**